### PR TITLE
Result code to be used for task failures to diagnose using error codes

### PR DESCRIPTION
**Pull request by nigurr**

### CODE REVIEW

File: src/Agent.Worker/TaskCommandExtension.cs [hunk 1]
```diff
@@ -573,6 +577,7 @@ internal static class TaskSetVariableEventProperties
     internal static class TaskCompleteEventProperties
     {
         public static readonly String Result = "result";
+        public static readonly String Code = "code";
```
Comment (TingluoHuang) [inline src/Agent.Worker/TaskCommandExtension.cs]: in what case you use the 'Code'?

File: src/Agent.Worker/TaskCommandExtension.cs [hunk 2]
```diff
@@ -435,6 +437,8 @@ private void ProcessTaskCompleteCommand(IExecutionContext context, Dictionary<st
             }
 
             context.Result = TaskResultUtil.MergeTaskResults(context.Result, result);
+            eventProperties.TryGetValue(TaskCompleteEventProperties.Code, out resultCode);
```
Comment (TingluoHuang) [inline src/Agent.Worker/TaskCommandExtension.cs]:
```C#
if(eventProperties.TryGetValue(TaskCompleteEventProperties.Code, out resultCode))
{
    context.ResultCode = resultCode;
}
```

File: src/Agent.Worker/TaskCommandExtension.cs (hunk 1 above)
Comment (nigurr) [inline src/Agent.Worker/TaskCommandExtension.cs]: As we discussed, for setting the task failure specific codes for different use cases customer issues/infra issues etc.

**Comment by nigurr** [comment]
@TingluoHuang Can I merge the PR now?

**Pull request merged**